src/zenml/integrations/neptune/experiment_trackers/run_state.py

```diff
@@ -0,0 +1,54 @@
+import neptune.new as neptune
+
+from zenml.client import Client
+from zenml.integrations.constants import NEPTUNE
+
+
+def singleton(class_):
+    instances = {}
+
+    def getinstance(*args, **kwargs):
+        if class_ not in instances:
+            instances[class_] = class_(*args, **kwargs)
+        return instances[class_]
+
+    return getinstance
+
+
+class NoActiveRunException(Exception):
+    pass
+
+
+@singleton
+class RunState:
+    def __init__(self):
+        self._run_id = ""  # potentially redundant
+        self._active_run = None
+
+    def set_active_run(
+        self, neptune_run: neptune.metadata_containers.Run
+    ) -> None:
+        self._active_run = neptune_run
+
+    @property
+    def active_run(self) -> neptune.metadata_containers.Run:
+        if self._active_run is None:
+            raise NoActiveRunException("No active run at the moment")
+        return self._active_run
+
+
+def get_neptune_run() -> neptune.metadata_containers.Run:
+    client = Client()
+    experiment_tracker = client.active_stack.experiment_tracker
+    if experiment_tracker.flavor == NEPTUNE:
+        return experiment_tracker.run_state.active_run
```

Comment:
Could we add handling of the `else` clause? Maybe an error?

Reply:
I think that error would be a better choice here, as it reflects the situation when a user tries to fetch a neptune run, while using a different experiment tracker. What do you think?

Reply:
Agree.